pkg/envtest/binaries.go

```diff
@@ -124,15 +124,54 @@ func downloadBinaryAssets(ctx context.Context, binaryAssetsDirectory, binaryAsse
 		}
 	}
 
+	var requestedRange semver.Range
+	if binaryAssetsVersion != "" {
```

Comment:
Do we really have to support all of this? I intentionally didn't duplicate all the messiness of setup-envtest when implementing this functionality here

Reply:
Independent of how much we support, let's extract this logic into a separate func and table test all cases

Reply:
No problem!